OpenRA.Mods.Common/Traits/Demolishable.cs

```diff
@@ -38,14 +45,35 @@ public bool PreventsEjectOnDeath(Actor self)
 			return info.PreventsEjectOnDeath;
 		}
 
-		public void Demolish(Actor self, Actor saboteur)
+		public void Demolish(Actor self, Actor saboteur, int delay)
 		{
-			self.Kill(saboteur);
+			triggerDemolition = true;
```

Comment:
This looks a bit dodgy, I'd expect to see a lock or other guard to prevent this from being called multiple times.